plugins/filter/lists.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.common.collections import is_sequence
+
+
+def remove_duplicates(lst):
+    seen = set()
```

Comment:
This doesn't work with non-hashable objects (like lists or dicts). You get something like `TypeError: unhashable type: 'list'`.

For that reason, ansible-core's `unique` code has a fallback in case of `TypeError`s, where it uses a *list* `seen` instead of a set. This is obviously slower, but it always works.